src/Util/ClassSourceManipulator.php

```diff
@@ -94,7 +94,12 @@ public function addEntityField(string $propertyName, array $columnOptions, array
         if ('array' === $typeHint) {
             $defaultValue = new Node\Expr\Array_([], ['kind' => Node\Expr\Array_::KIND_SHORT]);
         }
-        $this->addProperty($propertyName, $comments, $defaultValue);
+
+        if ($isTyped) {
+            $this->addProperty($propertyName, $comments, $defaultValue, $typeHint);
+        } else {
+            $this->addProperty($propertyName, $comments, $defaultValue);
+        }
```

Comment:
You don't need to repeat `addProperty` twice there:
```suggestion
        $this->addProperty($propertyName, $comments, $defaultValue, $isTyped ? $typeHint : null);
```